js/liqui.js

```diff
@@ -508,35 +508,31 @@ module.exports = class liqui extends Exchange {
         return this.orders[id];
     }
 
-    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
-        // if (!symbol)
-        //     throw new ExchangeError (this.id + ' fetchOrders requires a symbol');
-        await this.loadMarkets ();
-        let request = {};
-        let market = undefined;
-        if (symbol) {
-            let market = this.market (symbol);
-            request['pair'] = market['id'];
-        }
-        let response = await this.privatePostActiveOrders (this.extend (request, params));
-        let openOrders = [];
-        if ('return' in response)
-            openOrders = this.parseOrders (response['return'], market);
+    updateOrdersCache (openOrders, symbol) {
         for (let j = 0; j < openOrders.length; j++) {
-            this.orders[openOrders[j]['id']] = openOrders[j];
+            // update local cache with opened orders
+            const id = openOrders[j]['id'];
+            this.orders[id] = openOrders[j];
         }
         let openOrdersIndexedById = this.indexBy (openOrders, 'id');
         let cachedOrderIds = Object.keys (this.orders);
-        let result = [];
         for (let k = 0; k < cachedOrderIds.length; k++) {
+            // try to find every cached order in open orders array
+            // possible reasons why cached order is not there:
+            // - order was closed or cancelled -> update cache
+            // - symbol mismatch (e.g. cached BTC/USDT, fetched ETH/USDT) -> skip
             let id = cachedOrderIds[k];
-            if (id in openOrdersIndexedById) {
-                this.orders[id] = this.extend (this.orders[id], openOrdersIndexedById[id]);
-            } else {
+            if (!(id in openOrdersIndexedById)) {
+                // cached order is not in open orders array
                 let order = this.orders[id];
+                if (typeof symbol !== 'undefined' && symbol !== order['symbol']) {
+                    // fetched a particular symbol but it doesn't match the order in the cache -> won't update
+                    continue;
+                }
+                // order is cached but not opened -> mark as closed
                 if (order['status'] === 'open') {
                     order = this.extend (order, {
-                        'status': 'closed',
+                        'status': 'closed', // likewise it might have been canceled
```

Comment:
Yep, but only if the order wasn't closed by this instance of Liqui. I mean, when we cancelOrder, we cache the canceled status as well, so upon fetchOrders we don't enter this codepath at all. So, there's no way for the order status to change, unless it's changed externally. The cache has to be properly saved and restored between the runs to sustain it, of course.

Reply:
You're right. I indeed used different sessions.